buildbot/buildbot/worker.py

```diff
@@ -102,16 +102,20 @@ def run(self, cmd, log_stdout=True, timeout=60, cwd='', **kwargs):
 
 
 @contextmanager
-def work(db, old_state, temp_state, done_state):
+def work(db, old_state, temp_state, done_state_or_func):
     """A context manager for acquiring a job temporarily in an
     exclusive way to work on it. Produce a `JobTask`.
     Done state can either be a valid state string or a function that
     accepts a Task object and returns a valid state string.
     """
+    done_state = None
```

Comment:
This line isn't necessary. :smiley: And a slightly better name for this might be `done_func`, since it's a function either way now.

Reply:
Is there no way to disable magic hoisting? I don't like variables being pulled up from `if` scope.

Reply:
Only functions introduce scopes in Python, so you'd need to "thunk" the `if`.

Reply:
Facepalm. Okay, then let me keep the variable there for my own sanity. I've changed the name.